### ISSUE TITLE
Default preferences for databases

### ISSUE THREAD

**Issue by PositivePaulo** [strongbox-password-safe/Strongbox#332]
Hello @mmcguill,

I saw the comment on the [latest blog post](https://strongboxsafe.com/new-audit-feature/) of the Strongbox web site and have thought a bit about its second point. I am answering here so that it may be tracked effectively and as early as possible.

> I’d like the ability to save my view and audit preferences and make them the default for any new database that I load.

I think of three ways of achieving this:
 - What you suggested: importing settings from a previous database when creating a new one. This would sort of solve the problem, but I think not having a constant common basis for those preferences can be frustrating as well because one may not define a default set of values that changes independently of particular databases', especially if you delete all of the databases. Thus the following other two possibilities:
 - Having a default set of preferences that is copied in a new database upon creation. This would help because the import is from something that is not a database and so the default options can be kept separately from the rest.
 - Having a default set of preferences that the user may override on a case-by-case basis by modifying a database's particular settings. Only the modified, non-default settings would be saved in the database's entry, the rest would be loaded from the global set at run-time. This would enable the same as the previous possibility, but with the additional ability to change some settings once for all the databases not overriding them.

Let me know what you think.
Cheers,
Paul.

**Comment by strongbox-mark** [comment]
Thanks for the comment/issue @PositivePaulo - Yes, I think you're probably right and I think a combination of your 2nd and 3rd points is probably the way to go.

In fact there is already an implicit set of default preferences within Strongbox. So we're half way there. 

The real innovation is to store them with the database so that they get picked up automatically in Strongbox and an ability to independently export them if people want so they can set them as their defaults.

**Comment by PositivePaulo** [comment]
Glad I could help in some way!
If a basis for default preferences is already there, then it is even better.
Good idea, indeed. When I proposed those designs, I did not know you could consider modifying the database itself in order to achieve the desired result. I thought this client did not embed self-specific information, but if it already does and that it does not break other KeePass and Password Safe clients, then yes it would definitely be a good feature, especially when synchronizing with the macOS client.

**Comment by strongbox-mark** [comment]
Yes, well, at the moment Strongbox doesn't store any custom data in there, but it is a possibility and would be pretty handy for this kind of per database non-default preferences. Means it would just work on any of your devices straight away.

There's a custom data field area plugins and the like can use in the KDBX4 format which I think would be where this would be best suited and it _shouldn't_ break any clients as that's what it's designed for.
